### PR TITLE
[FileProcessor] Run untill the file is fixed completelly

### PR DESCRIPTION
**Pull request by TomasVotruba**
See https://twitter.com/rectorphp/status/1415102116725731330

E6NzoXMWUAUiCeX

### CODE REVIEW

**Comment by samsonasik** [comment]
@TomasVotruba it seems Github action is hit the account/organization concurrency limit?  https://github.com/rectorphp/rector-src/runs/3070057847

```bash
Can't find any online and idle self-hosted or hosted runner in the current repository, account/organization that matches the required labels: 'ubuntu-latest'
Found online and busy hosted runner(s) in the current repository's organization account that matches the required labels: 'ubuntu-latest'. Hit concurrency limits on the hosted runners. Waiting for one of them to get assigned for this job.
Waiting for a hosted runner in 'organization' to pick this job...
```

**Comment by TomasVotruba** [comment]
Yea, that's because this upgrade added temporarily infinite loops to few rules.
It's fixed now, but it has to cool down I presume :) I think in 12 hours it will ok

File: src/Application/FileProcessor/PhpFileProcessor.php [hunk 1]
```diff
@@ -49,24 +49,32 @@ public function process(File $file, Configuration $configuration): void
...
 
-        if ($file->hasErrors()) {
-            // cannot print file with errors, as print would b
-            $this->notifyPhase($file, ApplicationPhase::PRINT_SKIP());
-            return;
-        }
+            $this->tryCatchWrapper($file, function (File $file) use ($configuration): void {
+                $this->printFile($file, $configuration);
+            }, ApplicationPhase::PRINT());
+        } while ($file->hasChanged());
```
Comment (samsonasik) [inline src/Application/FileProcessor/PhpFileProcessor.php:77]: @TomasVotruba it seems make hanging on tests on packages tests, ref https://github.com/rectorphp/rector/issues/6576 

Add `break` before

```diff 
+      break;
    } while ( $file->hasChanged());
```

seems solve it on symfony-packages tests, but not sure if needs another tweak like check rector has completed the check.

Comment (samsonasik) [reply src/Application/FileProcessor/PhpFileProcessor.php:77]: Ok, add `break` cause error in current `tests` for multi rector:

```bash
1) Rector\Core\Tests\Issues\DoubleRun\DoubleRunTest::test with data set #0 (Symplify\SmartFileSystem\SmartFileInfo Object (...))
tests/Issues/DoubleRun/Fixture/fixture.php.inc
Failed asserting that string matches format description.
--- Expected
+++ Actual
@@ @@
 
 final class RemoveAll
 {
+    public function __construct()
+    {
+    }
 }
 
 ?>
 ```
 
 so it need a something "status" when the rector should stop running

**Pull request merged**